src/vim.js

```diff
@@ -1148,19 +1143,59 @@ export function initVim(CodeMirror) {
       }
     }
 
+    var specialKey = {
+      Return: 'CR', Backspace: 'BS', 'Delete': 'Del', Escape: 'Esc', Insert: 'Ins',
+      ArrowLeft: 'Left', ArrowRight: 'Right', ArrowUp: 'Up', ArrowDown: 'Down',
+      Enter: 'CR', ' ': 'Space'
+    };
+    var ignoredKeys = { Shift: 1, Alt: 1, Command: 1, Control: 1,
+      CapsLock: 1, AltGraph: 1, Dead: 1, Unidentified: 1 };
+    
+    function vimKeyFromEvent(e, vim) {
+      var key = e.key;
+      if (ignoredKeys[key]) return;
+      if (key.length > 1 && key[0] == "n") {
+        key = key.replace("Numpad", "");
+      }
+      key = specialKey[key] || key;
+
+      var name = '';
+      if (e.ctrlKey) { name += 'C-'; }
+      if (e.altKey) { name += 'A-'; }
+      if (e.metaKey) { name += 'M-'; }
+      // on mac many characters are entered as option- combos
+      // (e.g. on swiss keyboard { is option-8)
+      // so we ignore lonely A- modifier for keypress event on mac
+      if (CodeMirror.isMac && e.altKey && !e.metaKey && !e.ctrlKey) {
+        name = name.slice(2);
+      }
+      if ((name || key.length > 1) && e.shiftKey) { name += 'S-'; }
+  
+      if (!vim.expectLiteralNext && key.length == 1) {
+        if (langmap.keymap && key in langmap.keymap) {
+          if (langmap.remapCtrl || !name)
+            key = langmap.keymap[key];
+        } else if (key.charCodeAt(0) > 255) {
```

Comment:
The code responsible for automatic remapping of non-ascii characters is here. 

Removing this check should make it work for Dvorak too, but unlike non-latin case where we do not create any conflicts by automatically translating keys, for Dvorak so we would indeed have to add an opt-in setting, which does not exist in vim. 

The issue with "йй" may be solved by removing [length==1](https://github.com/replit/codemirror-vim/blob/81fafd426450b9dcc0d60923fbae4c088d41bb40/src/index.ts#L309C17-L309C34) check, but i am not sure what we can do to handle the dead key itself. Afaik browsers do not provide key code for them.

Reply:
what is the issue with "йй"? isn't that two separate letters?

Reply:
It is two separate letters, but I'm not in insert mode when typing them. Pressing the corresponding key (j) on the mnemonic keyboard layout in normal mode twice does not move the cursor down by two lines as expected, but rather inputs "йй" as if insert mode was being used.

I'm haven't researched the cause yet.

Reply:
It is two separate letters, but i'm not in insert mode when typing them. Pressing the corresponding key (j) on the mnemonic keyboard in normal mode doesn't move the cursor down by two lines as expected, but inserts йй as if insert mode was active.

Reply:
`йй`  is inserted because of `text.length == 1` at https://github.com/replit/codemirror-vim/blob/master/src/index.ts#L309. We should change it to iteration over characters, but interaction with spellcheck and Japanese ime, makes that a bit large change for this pull request.